google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -892,7 +892,7 @@ private PooledSession take() throws SpannerException {
             return s.session;
           }
         } catch (Exception e) {
-          TraceUtil.endSpanWithFailure(tracer.getCurrentSpan(), e);
```

Comment:
`tracer.getCurrentSpan()` is called outside of scope, which means `endSpanWithFailure` will execute on either READ_WRITE_TRANSACTION or READ_ONLY_TRANSACTION span instead of WAIT_FOR_SESSION span.